src/mlpack/methods/dbscan/dbscan_impl.hpp

```diff
@@ -107,14 +143,32 @@ size_t DBSCAN<RangeSearchType, PointSelectionPolicy>::Cluster(
 
   // Now set assignments.
   assignments.set_size(data.n_cols);
+  #ifdef MLPACK_USE_OPENMP
+  #pragma omp parallel for
+  #endif
   for (size_t i = 0; i < data.n_cols; ++i)
     assignments[i] = uf.Find(i);
```

Comment:
I don't think `UnionFind::Find()` is thread-safe, so I'm not sure that this can work correctly.  Or did I overlook something?